File: 704. Binary Search.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l = 0, h =  nums.Length - 1;
        while(l<=h){
            int p = (l+h)/2;
            if(nums[p] == target)return p;
            else if(nums[p] < target)l = p+1;
            else h = p-1;
        }
        return -1;
    }
}

File: 704. Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        int a = 0, b = nums.length - 1;
        int mid;
        while (a <= b) {
            mid = (a + b) / 2;
            if (nums[mid] == target) return mid;
            if (nums[mid] < target) 
                a = mid + 1;
            else
                b = mid - 1;
            
        }
        return -1;
    }
}
